sympy/utilities/tests/test_code_quality.py

```diff
@@ -184,6 +199,20 @@ def test_this_file(fname, test_file):
             if func_is_re.search(line) and not test_file_re.search(fname):
                 assert False, message_func_is % (fname, idx + 1)
 
+            # XXX Python 2 unicode import test.
+            # May remove after deprecating python 2.7.
+            if idx in (0, 1):
+                match = encoding_header_re.match(line)
+                if match:
+                    if match[1].lower() == 'utf-8':
+                        has_coding_utf8 = True
+            try:
```

Comment:
How about something like this:

    if not has_coding_utf8:
        try:
            line.encode(encoding='ascii')
        except UnicodeEncodeError:
            assert False, message_py2_unicode % (fname, idx + 1)